### PR TITLE
Add unit tests for common sidebar functionality

### PR DESCRIPTION
**Pull request by tgolen**
### Fixed Issues
This is part of the refactoring needed in this issue: https://github.com/Expensify/App/issues/7948

### Tests
The unit tests were based on the tests written in these PRs:
https://github.com/Expensify/App/pull/6084
https://github.com/Expensify/App/pull/5317

### QA Steps
Same as in these PRs:
https://github.com/Expensify/App/pull/6084
https://github.com/Expensify/App/pull/5317

cc @luacmartins

### CODE REVIEW

File: src/components/Icon/index.js [hunk 1]
```diff
@@ -41,27 +41,29 @@ class Icon extends PureComponent {
...
-                height={height}
-                fill={this.props.fill}
-            />
-        );
 
         if (this.props.inline) {
             return (
-                <View style={[StyleUtils.getWidthAndHeightStyle(width, height), styles.bgTransparent, styles.overflowVisible]}>
+                <View
+                    accessibilityHint={`${this.props.src.name} Icon`}
```
Comment (tgolen) [inline src/components/Icon/index.js:48]: This small refactoring done in this file was to add this prop, which allowed the Icon to be targetted by the unit tests. The rest of the changes in this file are just some cleaning up.

File: src/pages/home/sidebar/SidebarLinks.js [hunk 2]
```diff
@@ -48,6 +48,9 @@ const propTypes = {
 
         /** Number of unread actions on the report */
         unreadActionCount: PropTypes.number,
+
+        /** Whether or not the report has a draft comment */
```
Comment (luacmartins) [inline src/pages/home/sidebar/SidebarLinks.js:52]: NAB 
```suggestion
        /** Whether the report has a draft comment */
```

File: tests/unit/LHNOrderTest.js [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        REPORT_ACTIONS: 'reportActions_',
+    },
+};
+
+Onyx.init({
+    keys: ONYXKEYS,
+    registerStorageEventListener: () => {},
+});
+
+jest.disableAutomock();
```
Comment (luacmartins) [inline tests/unit/LHNOrderTest.js:123]: Can we add a comment explaining why we need this?

File: tests/unit/LHNOrderTest.js [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+        //             expect(pencilIcon).toHaveLength(1);
+        //
+        //             // console.log(sidebarLinks.toJSON().children[1].children[0].props.data[0].data)
+        //
+        //             // The reports should be in the order 3 > 2 > 1
+        //             const reportOptions = sidebarLinks.getAllByText(/ReportID, (One|Two|Three)/);
+        //             expect(reportOptions).toHaveLength(3);
+        //             expect(reportOptions[2].children[0].props.children).toBe('ReportID, One');
+        //         });
+        // });
```
Comment (luacmartins) [inline tests/unit/LHNOrderTest.js:279]: Can we create an issue for this instead of the comment?

**Review by luacmartins — COMMENTED** [review]
Left a few small comments

File: tests/unit/LHNOrderTest.js (hunk 3 above)
Comment (tgolen) [reply tests/unit/LHNOrderTest.js]: I'm going to remove it actually. I only added this here to see if it fixed some of the test errors, and it had no effect.

File: tests/unit/LHNOrderTest.js (hunk 4 above)
Comment (tgolen) [reply tests/unit/LHNOrderTest.js:277]: I've added it to this GH: https://github.com/Expensify/App/issues/7948

Comment (luacmartins) [reply tests/unit/LHNOrderTest.js:277]: Nice! Thanks for adding that.

**Review by luacmartins — APPROVED** [review]

**Review by danieldoglas — APPROVED** [review]
LGTM

**Pull request merged**